src/actions/navigation.js

```diff
@@ -27,17 +28,30 @@ export const chooseDisplayComponentFromURL = (url) => {
 
 /*
  * All the Fetch Promises are created before first render. When trying the cache we `await`.
- * If the Fetch is not finished, this will wait for it to end. Subsequent awaits will immeditaly return the result.
+ * If the Fetch is not finished, this will wait for it to end. Subsequent awaits will immediately return the result.
  * For the landing dataset, no problem either because await on a value just returns the value.
  */
 const tryCacheThenFetch = async (mainTreeName, secondTreeName, state) => {
```

Comment:
Should we rename this function to `awaitCachedDatasetPromise` or something more accurate? It doesn't fetch anything really if theres a cache miss, it just throws an error (since we shouldn't have cache misses if our cache-populating code is working the way I think it is).

Reply:
Agreed - I've changed it to `updateNarrativeDataset` and expanded it to handle the dispatch itself